packages/flutter/lib/src/material/menu_anchor.dart

```diff
@@ -923,6 +924,15 @@ class MenuItemButton extends StatefulWidget {
   /// {@endtemplate}
   final bool closeOnActivate;
 
+  /// The direction in which the menu item expands.
+  ///
+  /// If [menuDirection] is [Axis.vertical], the menu will be expanded vertically.
+  /// If [menuDirection] is [Axis.horizontal], then the menu  will be
+  /// expanded horizontally.
+  ///
+  /// Defaults to [Axis.vertical].
+  final Axis menuDirection;
```

Comment:
Thanks for your continued help with the fix! I'm actually thinking about the property name. Based on the doc and the name, it feels like setting `menuDirection: Axis.vertical` can control the menu to make it show up vertically. Not sure if that can cause any confusion. 

I'm now actually thinking maybe we don't need to add an extra property here. In `_MenuItemButtonState`, we can have a 

```dart
  _MenuAnchorState? get _anchor => _MenuAnchorState._maybeOf(context);
```

then we can just use `_anchor._orientation` to get the menu direction information(vertical or horizontal). Do you think if this makes sense?

Reply:
When writing test for this property,  I did write comments to make it clear how this property is adjusted for the layout to flex in certain direction.

```dart
      // Test a long MenuItemButton in an unconstrained layout with vertical menuDirection.
      await tester.pumpWidget(buildMenuButton(menuDirection: Axis.vertical));
      expect(tester.getSize(find.byType(MenuItemButton)), const Size(800.0, 48.0));

      // Test a long MenuItemButton in an unconstrained layout with horizontal menuDirection.
      await tester.pumpWidget(buildMenuButton(menuDirection: Axis.horizontal));
      expect(tester.getSize(find.byType(MenuItemButton)), const Size(800.0, 48.0));

      // Test a long MenuItemButton in a constrained layout with vertical menuDirection.
      await tester.pumpWidget(buildMenuButton(menuDirection: Axis.vertical, constrainedLayout: true));
      expect(tester.getSize(find.byType(MenuItemButton)), const Size(200.0, 120.0));

      // Test a long MenuItemButton in a constrained layout with horizontal menuDirection.
      await tester.pumpWidget(buildMenuButton(menuDirection: Axis.horizontal, constrainedLayout: true));
      expect(tester.getSize(find.byType(MenuItemButton)), const Size(200.0, 48.0));
      // This should throw an error.
      final AssertionError exception = tester.takeException() as AssertionError;
      expect(exception, isAssertionError);
```

You're absolutely right we need to make clear for the users to avoid confusion.

Reply:
> The _anchor._orientation is null when MenuItemButton is used directly.

I see! You are right. I'm thinking about 3 options. One is still getting the direction from anchor, but we need to explicitly mention in the doc that these buttons are just used for menus; Other use cases might cause overflow. Because I'm not sure whether *menu*ItemButton should be used without any menu involved🤣. 

The second option is just to give a better property name, `menuFlexDirection` is a good one, I'm also thinking about a boolean `allowHorizontalExpansion` or something else. If the button is placed on a vertical menu, it is false and we do clip to avoid overflow; if it is placed on a horizontal menu, we should allow the expansion. But for this option, it is still a breaking change for `MenuBar`s if they already use `MenuItemButton`. The default property value will cause exception and users need to explicitly assign a value to `menuFlexDirection`.

The third one is to combine the first two. We still expose this property, but if anchor is not null, the anchor direction should override the property to avoid breaking change.🙂 Let me know which one is preferred!

Reply:
Please also let me know if I missed any other PR review!🙂 Thanks a lot!

Reply:
I've changed the name and condition where `_anchor` orientational can override the property. This implemented the third option you suggest.
https://github.com/flutter/flutter/pull/143932/commits/01006f3a90802f32558345af2f98e893ddaa1b3f

Please let me know what do you think.